.github/workflows/pre-commit.yml

```diff
@@ -0,0 +1,14 @@
+name: pre-commit
+
+on:
+  pull_request:
+  push:
+    branches: [source]
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - uses: actions/setup-python@v3
+    - uses: pre-commit/action@v3.0.0
```

Comment:
This action is in maintenance mode. It is still maintained but they are advising using `pre-commit.ci`. Did you check that?

Reply:
I checked briefly. It looks like a full blown CI system with fancy stuff that I don't think we want. The tests would run through their SaaS and fix any issues that can be fixed automatically.

> if tools make changes to files during a pull request, pre-commit.ci will automatically fix the pull request.

I would say we can continue with the pre-commit action. It is maintained, but any new feature requests are basically directed to pre-commit.ci.

Reply:
Yeah it looked like that to me as well. It is still good to acknowledge that this action might not be maintained for too long.